### PR TITLE
Refactor EventStore::Memory

### PR DESCRIPTION
**Pull request by andrewgr**
Just minor refactorings.

### CODE REVIEW

File: lib/event_sourcery/event_store/memory.rb [hunk 1]
```diff
@@ -11,37 +11,34 @@ def initialize(events = [], event_builder: EventSourcery.config.event_builder)
...
         true
       end
 
       def get_next_from(id, event_types: nil, limit: 1000)
-        events = @events.select { |event| event.id >= id }
-        if event_types
-          events = events.select { |event| event_types.include?(event.type) }
-        end
-        events.first(limit)
+        events = event_types.nil? ? @events : @events.select { |e| event_types.include?(e.type) }
```
Comment (andrewgr) [inline lib/event_sourcery/event_store/memory.rb]: At first I extracted this line into a method but then it had a bit strange behaviour and a bit strange name IMO.

File: lib/event_sourcery/event_store/memory.rb [hunk 2]
```diff
@@ -11,37 +11,34 @@ def initialize(events = [], event_builder: EventSourcery.config.event_builder)
...
+        events.select { |event| event.id >= id }.first(limit)
```
Comment (twe4ked) [inline lib/event_sourcery/event_store/memory.rb]: I think I actually find this easier to parse on multiple lines.

File: lib/event_sourcery/event_store/memory.rb [hunk 3]
```diff
@@ -11,37 +11,34 @@ def initialize(events = [], event_builder: EventSourcery.config.event_builder)
...
-          id = @events.size + 1
-          serialized_body = EventBodySerializer.serialize(event.body)
           @events << @event_builder.build(
-            id: id,
+            id: @events.size + 1,
             aggregate_id: event.aggregate_id,
             type: event.type,
             version: next_version(event.aggregate_id),
-            body: serialized_body,
+            body: EventBodySerializer.serialize(event.body),
```
Comment (twe4ked) [inline lib/event_sourcery/event_store/memory.rb]: Not sure what we're gaining by inlining these variables.

Comment (andrewgr) [inline lib/event_sourcery/event_store/memory.rb]: Just a bit less code to read. Shorter method.

**Pull request merged**